### PR TITLE
Move master api haproxy to assets

### PR DESCRIPTION
**Pull request by celebdor**
As part of the move, haproxy.sh was made not to depend on environment
substitution.

It was also bash linted a bit

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/302/

File: assets/files/etc/haproxy/haproxy.cfg.template [hunk 1]
```diff
@@ -18,5 +18,5 @@ backend masters
    option external-check
    option log-health-checks
    external-check path "/usr/bin:/usr/local/bin"
-   external-check command "/usr/local/etc/haproxy/members_healthcheck.sh"
-   balance     roundrobin
\ No newline at end of file
+   external-check command "/etc/haproxy/members_healthcheck.sh"
```
Comment (yboaron) [inline assets/files/etc/haproxy/haproxy.cfg.template]: Checked it locally on my setup looks OK.
You should leave the external check command point to pod's directory, i.e
"/usr/local/etc/haproxy/members_healthcheck.sh"

With your change, we get [1] Error

[1] [ALERT] 078/120525 (36) : Failed to exec process for external health check: No such file or directory. Aborting.

File: assets/files/etc/haproxy/members_healthcheck.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec wget -q --no-check-certificate "https://$3:6443/healthz"
```
Comment (vrutkovs) [inline assets/files/etc/haproxy/members_healthcheck.sh]:
```suggestion
exec wget -q --no-check-certificate "https://$3:6443/readyz"
```

File: assets/files/usr/local/bin/haproxy.sh [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    local haproxy_cfg_ip
+    local domain
+    local cfg
+
+    domain="$1"
+    cfg="$2"
+    for item in $(etcd_members "$domain"); do
+        ip=$(first_a_addr "$item")
+        haproxy_cfg_ip=$(awk -v "server=$item" '$2 == server {print substr($3, 0, index($3, ":") - 1); exit}' "$cfg")
+        if [[ $? -eq 0 && -n "$ip" ]]; then
```
Comment (yboaron) [inline assets/files/usr/local/bin/haproxy.sh]: The '$? -eq 0' refers to the operation in line #48, right?
In what case, we'll get an error (!=0) for that line?

File: assets/templates/99_master-api-haproxy.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      units:
+        - contents: |
+            [Unit]
+            Description=Load Balance OpenShift API using HAProxy
+            Wants=network-online.target
+            After=network-online.target
+
+            [Service]
+            WorkingDirectory=/etc/haproxy
+            ExecStartPre=/usr/local/bin/haproxy.sh
```
Comment (yboaron) [inline assets/templates/99_master-api-haproxy.yaml]: Do we need this line?

Comment (vrutkovs) [reply assets/templates/99_master-api-haproxy.yaml]:
```suggestion
```

This causes
`Mar 20 12:29:33 master-0 haproxy.sh[3684]: Must provide parameter (start/stop)`

Comment (celebdor) [reply assets/templates/99_master-api-haproxy.yaml]: Done. Thanks

File: assets/files/etc/haproxy/members_healthcheck.sh (hunk 2 above)
Comment (celebdor) [reply assets/files/etc/haproxy/members_healthcheck.sh]: Done

Comment (celebdor) [reply assets/files/etc/haproxy/members_healthcheck.sh]: Finally this endpoint does not seem to be there

File: assets/files/etc/haproxy/haproxy.cfg.template [hunk 5]
```diff
@@ -15,8 +13,6 @@ frontend  main
   bind :7443
   default_backend masters
 backend masters
-   option external-check
+   option httpchk GET /healthz HTTP/1.0
```
Comment (yboaron) [inline assets/files/etc/haproxy/haproxy.cfg.template]: Checked with this haproxy.cfg in my env, 
All members are DOWN (health check fail)

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/309/

**Comment by celebdor** [comment]
Just tested it with success!

File: assets/files/usr/local/bin/haproxy.sh [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    return 1
+}
+
+function start_haproxy {
+    local domain
+    local api_vip
+
+    domain="$1"
+    api_vip="$2"
+    declare -r image=quay.io/yboaron/haproxy:latest
```
Comment (yboaron) [inline assets/files/usr/local/bin/haproxy.sh]: As we don't need the external-check anymore, we can use the official haproxy image
https://hub.docker.com/_/haproxy

Comment (celebdor) [reply assets/files/usr/local/bin/haproxy.sh]: Will do in a follow-up PR. Thanks for the pointer

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/317/

**Pull request merged**